mtls_server/server.py

```diff
@@ -72,13 +101,178 @@ def get_ca_cert():
 
     @app.route("/crl", methods=["GET"])
     def get_crl():
-        crl = handler.cert_processor.get_crl()
+        crl = cert_processor.get_crl()
         return crl.public_bytes(serialization.Encoding.PEM).decode("UTF-8")
 
     @app.route("/version", methods=["GET"])
     def get_version():
         return json.dumps({"version": version}), 200
 
+
+    @app.route("/certs", methods=["GET"])
+    def get_certificates():
+        return error_response("Not implemented", status_code=501)
+
+    @app.route("/certs", methods=["POST"])
+    @login_required
+    def create_cert():
+        body = request.get_json()
+        fingerprint = g.user_fingerprint
+        lifetime = int(body.get("lifetime"))
+        min_lifetime = Config.get_int("mtls", "min_lifetime", 60)
+        max_lifetime = Config.get_int("mtls", "max_lifetime", 0)
+        if lifetime < min_lifetime:
+            logger.info(
+                f"User requested lifetime less than minimum. {lifetime} < {min_lifetime}"
+            )
+            lifetime = min_lifetime
+        if max_lifetime != 0:
+            if lifetime > max_lifetime:
+                logger.info(
+                    f"User requested lifetime greater than maximum. {lifetime} < {max_lifetime}"
+                )
+                lifetime = max_lifetime
+        csr_str = body["csr"]
+        csr = cert_processor.get_csr(csr_str)
+        if csr is None:
+            return error_response("Could not load CSR")
+        cert = None
```

Comment:
Shouldn't there be a lot more validation of the `csr` here?

Reply:
https://github.com/drGrove/mtls-server/blob/refactor/api/mtls_server/cert_processor.py#L446

https://github.com/drGrove/mtls-server/blob/refactor/api/mtls_server/cert_processor.py#L452

Reply:
What http status code would those validation failures lead to?

Reply:
501 is the default

Reply:
validation failures should be a 4xx error; not a 501.